### ISSUE TITLE
Default value derived from function with optional output parameter broke first time migration

### ISSUE THREAD

**Issue by mkhlsavin** [edgedb/edgedb#6374]
- EdgeDB Version: 4.0-rc.1+fd5ed53
- OS Version: Win11

Steps to Reproduce:

1. Lets define function with optional output:
```
    function getDefaultCurrency() -> optional Currency {
        volatility:= 'Stable';
        using (
            assert_single((SELECT Currency FILTER .code IN global currentOrganization.currencies.code LIMIT 1)))
    }

```

2. Lets  define type with default property value:
```
    abstract type Currencied {
        required currency: Currency {
            readonly:= true;
            # Comment it on the first migration
            default:= getDefaultCurrency();
        };
    }
```

3. Lets create first time migration:
```
  CREATE FUNCTION default::getDefaultCurrency() -> OPTIONAL default::Currency {
      SET volatility := 'Stable';
      USING (std::assert_single((SELECT
          default::Currency FILTER
              (.code IN (GLOBAL default::currentOrganization).currencies.code)
      LIMIT
          1
      )))
  ;};
  ALTER TYPE default::Currencied {
      CREATE REQUIRED LINK currency: default::Currency {
          SET default := (default::getDefaultCurrency());
          SET readonly := true;
      };
  };
```

4. When you apply it, the bug is coming:
```
UnsupportedFeatureError: functions that reference globals may not be used when converting/populating data in migrations
    ┌─ C:\...\dbschema\migrations\00001.edgeql:896:7
    │
896 │ ╭       CREATE REQUIRED LINK currency: default::Currency {
897 │ │           SET default := (func::getDefaultCurrency());
898 │ │           SET readonly := true;
899 │ │       };
    │ ╰───────^ error

edgedb error: error in one of the migrations
```

The workaround are: comment the line in the first migration `SET default := (default::getDefaultCurrency());`, then create second migration and the bug disappear

**Comment by mkhlsavin** [comment]
Perhaps this is not a bug but expected behavior since default values can be used to seed properties. Then you need to describe a method for solving similar problems - for example, a flag that disables seeding during the first migration, etc.

**Comment by msullivan** [comment]
This ought to be allowed, though globals are always treated as empty while seeding the properties during the migration. It already works for bare uses of globals in the default (#5268), but I forgot to handle the function case.

I'll have a fix in 3.5
